File: OOP/Projekt04__konsolowe-OOP__klasa-Kosc/cpp/main.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <ctime>

using namespace std;

class Kosc
{
public:
    static unsigned int instanceCount;

public:
    string imageFiles[7] = {"kosc0.png", "kosc1.png", "kosc2.png", "kosc3.png", "kosc4.png", "kosc5.png", "kosc6.png"};
    unsigned int throwedNumber;
    int imageIndex;
    bool isKoscAvailble;

    Kosc(unsigned int userValue)
    {
        if (userValue > 6)
        {
            this->throwedNumber = 0;
            this->imageIndex = 0;
        }
        else
        {
            this->throwedNumber = userValue;
            this->imageIndex = userValue;
        };

        this->isKoscAvailble = true;
        Kosc::instanceCount += 1;
    };

    Kosc()
    {
        unsigned int value = (rand() % 6) + 1;
        this->imageIndex = value;
        this->throwedNumber = value;
        this->isKoscAvailble = true;
        Kosc::instanceCount += 1;
    }

    void throwKosc()
    {
        if (!this->isKoscAvailble)
            return;

        unsigned int value = (rand() % 6) + 1;
        this->imageIndex = value;
        this->throwedNumber = value;
        this->isKoscAvailble = true;
    }

    void blockKosc()
    {
        this->isKoscAvailble = false;
    }

    string valueMapper()
    {
        string mappedValues[7] = {"zero", "jeden", "dwa", "trzy", "cztery", "piec", "szesc"};

        return mappedValues[this->throwedNumber];
    }
};

unsigned int Kosc::instanceCount = 0;

void printInfo(Kosc &kosc)
{
    cout
        << "-------------------------------------" << endl;
    cout << "Liczba utworzonych instancji: " << kosc.instanceCount << endl;
    cout << "Wyrzucono: " << kosc.throwedNumber << " (" << kosc.valueMapper() << ")" << endl;
    cout << "Plik zdjeciowy: " << kosc.imageFiles[kosc.imageIndex] << endl;
}

int main()
{
    srand(time(0));

    unsigned int koscValue;

    cout << "Podaj wartosc nowej kosci: ";
    cin >> koscValue;

    Kosc k1(koscValue);
    printInfo(k1);

    Kosc k2;
    printInfo(k2);

    return 0;
}

File: OOP/Projekt04__konsolowe-OOP__klasa-Kosc/python/main.py
import random


class Kosc:
    instanceCount: int = 0

    def __init__(self, userValue: int = 0) -> None:
        self.imageFiles = [f"kosc{i}.png" for i in range(7)]
        if userValue == 0:
            value = random.randint(1, 6)
            self.imageIdx: int = value
            self.throwedValue: int = value
        elif userValue > 6:
            self.throwedValue: int = 0
            self.imageIdx: int = 0
        else:
            self.throwedValue: int = userValue
            self.imageIdx: int = userValue
        self.isKoscAvailble: bool = True
        Kosc.instanceCount += 1

    def throw(self) -> None:
        if not self.isKoscAvailble:
            return

        value = random.randint(1, 6)

        self.imageIdx = value
        self.throwedValue = value
        self.isKoscAvailble = True

    def block(self) -> None:
        self.isKoscAvailble = False

    def valueMapper(self) -> str:
        values = ["zero", "jeden", "dwa", "trzy", "cztery", "piec", "szesc"]

        return values[self.throwedValue]


def printKosc(kosc: Kosc) -> None:
    print("-------------------------------------")
    print(f"Liczba utworzonych instancji: {kosc.instanceCount}")
    print(f"Wyrzucono: {kosc.throwedValue} ({kosc.valueMapper()})")
    print(f"Plik zdjęciowy: {kosc.imageFiles[kosc.imageIdx]}")


def main():
    koscValue = int(input("Podaj wartosc nowej kosci: "))

    k1 = Kosc(koscValue)
    printKosc(k1)
    k2 = Kosc()
    printKosc(k2)


if __name__ == "__main__":
    main()
